src/backends/github/API.js

```diff
@@ -282,94 +237,70 @@ export default class API {
-      .then(changeTree => this.commit(options.commitMessage, changeTree))
-      .then(commitResponse => this.createBranch(branchName, commitResponse.sha))
-      .then(branchResponse => this.createPR(options.commitMessage, branchName))
-      .then(prResponse => this.user().then(user => user.name ? user.name : user.login)
-        .then(username => this.storeMetadata(contentKey, {
-          type: "PR",
-          pr: {
-            number: prResponse.number,
-            head: prResponse.head && prResponse.head.sha,
-          },
-          user: username,
-          status: status.first(),
-          branch: branchName,
-          collection: options.collectionName,
-          title: options.parsedData && options.parsedData.title,
-          description: options.parsedData && options.parsedData.description,
-          objects: {
-            entry: {
-              path: entry.path,
-              sha: entry.sha,
-            },
-            files: filesList,
-          },
-          timeStamp: new Date().toISOString(),
-        }
-        )));
-    } else {
-      // Entry is already on editorial review workflow - just update metadata and commit to existing branch
-      return this.getBranch(branchName)
+    const commitPromise = this.getBranch()
       .then(branchData => this.updateTree(branchData.commit.sha, "/", fileTree))
-      .then(changeTree => this.commit(options.commitMessage, changeTree))
-      .then((response) => {
-        const contentKey = entry.slug;
-        const branchName = `cms/${ contentKey }`;
-        return this.user().then(user => user.name ? user.name : user.login)
-        .then(username => this.retrieveMetadata(contentKey))
-        .then((metadata) => {
-          let files = metadata.objects && metadata.objects.files || [];
-          files = files.concat(filesList);
-          const updatedPR = metadata.pr;
-          updatedPR.head = response.sha;
-          return {
-            ...metadata,
-            pr: updatedPR,
-            title: options.parsedData && options.parsedData.title,
-            description: options.parsedData && options.parsedData.description,
-            objects: {
-              entry: {
-                path: entry.path,
-                sha: entry.sha,
-              },
-              files: _.uniq(files),
-            },
-            timeStamp: new Date().toISOString(),
-          };
-        })
-        .then(updatedMetadata => this.storeMetadata(contentKey, updatedMetadata))
-        .then(this.patchBranch(branchName, response.sha));
-      });
-    }
+      .then(changeTree => this.commit(options.commitMessage, changeTree));
+
+    const usernamePromise = this.user().then(user => (user.name ? user.name : user.login));
+
+    const initialMetadata = {
+      title: options.parsedData && options.parsedData.title,
+      description: options.parsedData && options.parsedData.description,
+      timeStamp: new Date().toISOString(),
+    };
+
+    return (unpublished
```

Comment:
A lot of this really should be broken down and notated for clarity at some point in the future.